### PR TITLE
Clean up/simplify auth

### PR DESCRIPTION
**Pull request by mmcgahan**
While writing up the documentation for the `requestAuthPlugin`, I came across a few elements of the auth flow that could be simplified. These are those changes.

### CODE REVIEW

**Comment by mmcgahan** [comment]
hm looks like this branch includes #42 as well

**Comment by mmcgahan** [comment]
there we go. much better

**Comment by mattkime** [comment]
lgtm

File: src/epics/sync.test.js [hunk 1]
```diff
@@ -43,6 +43,7 @@ describe('Sync epic', () => {
 
 		const pathname = '/noQuery';
 		const noMatchLocation = { ...MOCK_RENDERPROPS.location, pathname };
+		console.log(noMatchLocation.query, 'asdfasdf');
```
Comment (mattkime) [inline src/epics/sync.test.js]: mistaken commit?

Comment (mmcgahan) [inline src/epics/sync.test.js]: ooh, yup good catch

**Comment by sadafie** [comment]
LGTM

**Pull request merged**